app/controllers/rocket_job_mission_control/jobs_controller.rb

```diff
@@ -133,9 +133,88 @@ def edit
       authorize! :edit, @job
     end
 
-    def exceptions
-      authorize! :read, @job
-      @exceptions = @job.input.group_exceptions
+    def view_slice
+      # Params from RocketJob. Exceptions are grouped by class_name.
+      # Scope: [[slice1], [slice2], [slice(n)]
+      authorize! :view_slice, @job
+      error_type = params[:error_type]
+      scope      = @job.input.failed.where('exception.class_name' => error_type)
+
+      # Used by pagination to display the correct slice
+      # Offset refers to the slice number from the array "scope".
+      @offset         = params.fetch(:offset, 0).to_i
+      current_failure = scope.order(_id: 1).limit(1).skip(@offset).first
+
+      # Instance variables to share with the view and pagination.
+      @lines                 = current_failure.records
+      @failure_exception     = current_failure.try!(:exception)
+      @view_slice_pagination = {
+        record_number: current_failure['exception']['record_number'],
+        offset:        @offset,
+        total:         (scope.count - 1)
+      }
+    end
+
+    def edit_slice
+      # We need all the instance varaibles from the view_slice (above) to able to
+      # Build the form as an array but only display the bad line
+      authorize! :edit_slice, @job
+      error_type         = params[:error_type]
+      @line_index        = params[:line_index].to_i
+      @offset            = params.fetch(:offset, 0).to_i
+      scope              = @job.input.failed.where('exception.class_name' => error_type)
+      current_failure    = scope.order(_id: 1).limit(1).skip(@offset).first
+      @lines             = current_failure.records
+      @failure_exception = current_failure.try!(:exception)
+    end
+
+    def update_slice
+      authorize! :update_slice, @job
+
+      # Params from the edit_slice form
+      error_type      = params[:error_type]
+      offset          = params[:offset]
+      updated_records = params['job']['records']
+
+      # Finds specific slice [Array]
+      slice         = @job.input.failed.skip(offset).first
+
+      # Assings modified slice (from the form) back to slice
+      slice.records = updated_records
+
+      if slice.save!
```

Comment:
```suggestion
      if slice.save
```

`save!` will raise an exception if the save fails, so the else clause would never be executed.